applications/asset_tracker_v2/src/events/app_module_event.c

```diff
@@ -56,13 +56,25 @@ static char *get_evt_type_str(enum app_module_event_type type)
 static void log_event(const struct app_event_header *aeh)
 {
 	const struct app_module_event *event = cast_app_module_event(aeh);
-	char data_types[50] = "\0";
+	char data_types[60] = "\0";
 
 	if (event->type == APP_EVT_ERROR) {
 		APP_EVENT_MANAGER_LOG(aeh, "%s - Error code %d",
 				get_evt_type_str(event->type), event->data.err);
 	} else if (event->type == APP_EVT_DATA_GET) {
 		for (int i = 0; i < event->count; i++) {
+			/* data_types should contain space for the stringified type. */
+			size_t max_append_len = sizeof(data_types) - strlen(data_types);
+
+			if (i < event->count - 1) {
+				/* For every type except the last one, we also need space
+				 * for a comma and a space character.
+				 */
+				max_append_len = max_append_len - 2;
+			}
+			if (strlen(type2str(event->data_list[i])) > max_append_len) {
```

Comment:
```suggestion
			if (strlen(type2str(event->data_list[i])) >= max_append_len) {
```

What happens if the `data_list` item string length is equal to `max_append_len`? `strcat` includes the null character when it copies from the source buffer. This could cause `strcat` to write past the destination buffer.

```
    char data_types[5] = "\0";
    char *some_string = "12345";
    strcat(data_types, some_string);
    printf("%s\n", data_types);
```
prints "12345" which indicates a buffer size of 6.

Reply:
agreed. I think its then better to reduce in 1 in the calculation of the `max_append_len` in line 67. That way the `max_append_len` stays true to its name. I will push a change now. See what you think